backend/coins/eth/etherscan/etherscan.go

```diff
@@ -50,17 +52,22 @@ const ERC20GasErr = "insufficient funds for gas * price + value"
 type EtherScan struct {
 	url        string
 	httpClient *http.Client
+	limiter    *rate.Limiter
 }
 
 // NewEtherScan creates a new instance of EtherScan.
 func NewEtherScan(url string, httpClient *http.Client) *EtherScan {
 	return &EtherScan{
 		url:        url,
 		httpClient: httpClient,
+		limiter:    rate.NewLimiter(rate.Limit(CallsPerSec), 1),
 	}
 }
 
 func (etherScan *EtherScan) call(params url.Values, result interface{}) error {
+	if err := etherScan.limiter.Wait(context.TODO()); err != nil {
```

Comment:
`Wait` needs a `context` that we do not have here. I'm open to suggestions. FWIW we didn't have it before either